src/propstepsolvers.jl

```diff
@@ -34,48 +34,48 @@ QuKrylov() = QuKrylov(@compat Dict(:NC=>10))
 Propagates to the next time state
 Input Parameters:
 `prob`             :  Method to be used
-`op`               :  Hamiltonian of the system
+`eq`               :  Quantum Equation type
 `t`                :  Time corresponding to the t_state
 `current_t`        :  Current time
 `current_qustate`  :  Quantum state corresponding to current time
 """ ->
-function propagate(prob::QuEuler, op, t, current_t, current_qustate)
+function propagate(prob::QuEuler, eq::QuEquation, t, current_t, current_qustate)
     dt = t - current_t
-    return (eye(op)-im*op*dt)*current_qustate
+    return (eye(operator(eq))-im*operator(eq)*dt)*vec(current_qustate)
 end
 
-function propagate(prob::QuCrankNicolson, op, t, current_t, current_qustate)
+function propagate(prob::QuCrankNicolson, eq::QuEquation, t, current_t, current_qustate)
     dt = t - current_t
-    uni = eye(op)-im*op*dt/2
-    return \(uni', uni*current_qustate)
+    uni = eye(operator(eq))-im*operator(eq)*dt/2
+    return \(uni', uni*vec(current_qustate))
 end
 
-function propagate(prob::QuKrylov, op, t, current_t, current_qustate)
+function propagate(prob::QuKrylov, eq::QuEquation, t, current_t, current_qustate)
     dt = t - current_t
-    basis_size = get(prob.options,:NC, length(current_qustate))
-    N = min(basis_size, length(current_qustate))
-    v = Array(typeof(current_qustate),0)
+    basis_size = get(prob.options,:NC, length(vec(current_qustate)))
+    N = min(basis_size, length(vec(current_qustate)))
+    v = Array(typeof(vec(current_qustate)),0)
     @compat sizehint!(v, N+1)
-    push!(v,zeros(current_qustate))
-    push!(v,current_qustate)
+    push!(v,zeros(vec(current_qustate)))
+    push!(v,vec(current_qustate))
     alpha = Array(Complex{Float64},0)
     @compat sizehint!(alpha, N)
     beta = Array(Complex{Float64},0)
     @compat sizehint!(beta, N+1)
     push!(beta,0.)
     for i=2:N
-        w = op*v[i]
+        w = operator(eq)*v[i]
```

Comment:
Similarly, it would be good to have `op=operator(eq)` before the loop (since we can't assume that `operator` won't be expensive).